service/slice_config_webhook_validation.go

```diff
@@ -397,6 +397,10 @@ func preventUpdate(ctx context.Context, sc *controllerv1alpha1.SliceConfig, old
 		if exists && new.Protocol != oldType.Protocol {
 			return field.Forbidden(field.NewPath("Spec").Child("SliceGatewayProvider").Child("SliceGatewayServiceType"), "updating gateway protocol is not allowed")
 		}
+		// when no protocol is specified then it's assumed to be UDP, so can't be updated to TCP
+		if !exists && new.Protocol != defaultSliceGatewayServiceProtocol {
```

Comment:
Have you tested this out by adding a new cluster to the list of clusters and also setting the gw svc type for it at the same time? I am not sure if that would work. You should have to first check if the cluster exists in the old config. If it does not exist then this new conditional would not allow you to set the proto to TCP for the new cluster. This conditional should be checked only if the old config contains the cluster. Can you confirm?

Reply:
checking

Reply:
Made required changes to check the proto only if the old config contains the cluster